osbuilder: the dracut method is broken

When building the osbuilder with dracut method, it fails because the to be installed version of rust isn't calculated.

Here is the error log:
```
INFO: Create /go/src/github.com/kata-containers/kata-containers/tools/osbuilder/dracut_overlay/etc
INFO: Configure chrony file /go/src/github.com/kata-containers/kata-containers/tools/osbuilder/dracut_overlay/etc/chrony.conf
/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/rootfs.sh: line 555: rustup: command not found
/go/src/github.com/kata-containers/tests /go/src/github.com/kata-containers/kata-containers/tools/osbuilder
Install rust null
info: downloading installer
error: Pre-checks for host and toolchain failed: invalid toolchain name: 'null'
If you are unsure of suitable values, the 'stable' toolchain is the default.
Valid host triples look something like: x86_64-unknown-linux-gnu
[install_rust.sh:25] ERROR: sh -s -- -y --default-toolchain null
Failed at 555: bash ${script_dir}/../../../ci/install_rust.sh ${RUST_VERSION}
make: *** [Makefile:153: /go/src/github.com/kata-containers/kata-containers/tools/osbuilder/dracut_overlay] Error 1
[install_kata_image.sh:54] ERROR: sudo -E BUILD_METHOD=dracut make -e initrd
[build_install.sh:77] ERROR: .ci/openshift-ci/../install_kata_image.sh
```
Notice it is being passed `null` to the `install_rust.sh` script. This regression was introduced by commit f34f67d610146cb954.

--- a/tools/osbuilder/rootfs-builder/rootfs.sh
+++ b/tools/osbuilder/rootfs-builder/rootfs.sh
@@ -552,7 +552,13 @@ EOT
 		fi
 		[ "$LIBC" == "musl" ] && bash ${script_dir}/../../../ci/install_musl.sh
 		# rust agent needs ${arch}-unknown-linux-${LIBC}
-		rustup show | grep linux-${LIBC} > /dev/null || bash ${script_dir}/../../../ci/install_rust.sh ${RUST_VERSION}
+		if ! (rustup show | grep -v linux-${LIBC} > /dev/null); then
+			if [ "$RUST_VERSION" == "null" ]; then
+				detect_rust_version || \
+					die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-main}'."
+			fi
+			bash ${script_dir}/../../../ci/install_rust.sh ${RUST_VERSION}
+		fi
 		test -r "${HOME}/.cargo/env" && source "${HOME}/.cargo/env"
 		[ "$ARCH" == "aarch64" ] && OLD_PATH=$PATH && export PATH=$PATH:/usr/local/musl/bin
 

